clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -560,21 +513,25 @@ void ConstraintBuilderConsumer::HandleTranslationUnit(ASTContext &C) {
     else
       errs() << "Analyzing\n";
   }
-  TypeVariableBindingsMapT TypeVariableBindings;
-  GlobalVisitor GV = GlobalVisitor(&C, Info, TypeVariableBindings);
+
+
+  VariableAdderVisitor VAV = VariableAdderVisitor(&C, Info);
+  TypeVarVisitor TV = TypeVarVisitor(&C, Info);
+  GlobalVisitor GV = GlobalVisitor(&C, Info, TV);
   TranslationUnitDecl *TUD = C.getTranslationUnitDecl();
   // Generate constraints.
   for (const auto &D : TUD->decls()) {
```

Comment:
If the order matters, say that here.

Reply:
Yep, order is very important. I'll add a comment.